backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraSingleton.java

```diff
@@ -0,0 +1,53 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.backends.cassandra;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.james.backends.cassandra.components.CassandraModule;
+import org.apache.james.util.Host;
+
+public class CassandraSingleton {
+    public static DockerCassandraRule dockerCassandra = new DockerCassandraRule();
+
+    private static final int maxItemCount = 1000;
+    private static final AtomicInteger pastCreatedItemCount = new AtomicInteger(0);
+
+    static {
+        dockerCassandra.start();
+    }
+
+    public static Host getManagedHost(CassandraModule module) {
+        int itemCount = module.moduleTables().size() + module.moduleTypes().size();
+
+        if (pastCreatedItemCount.addAndGet(itemCount) > maxItemCount) {
```

Comment:
I don't understand why counting items is needed, can you extract this part and add a little comment?

Reply:
Because the number of test run is not significant for container restart, but number of created tables is ( 300+ test runs for DAO is really OK, but 60+ full server run might fail...)